myriad-scheduler/src/main/java/org/apache/myriad/configuration/NodeManagerConfiguration.java

```diff
@@ -91,6 +102,18 @@ public Double getCpus() {
     return Optional.fromNullable(cpus).or(DEFAULT_NM_CPUS);
   }
 
+  public synchronized Map<String, Long> getPorts() {
+    if (ports == null) {
+      //Good idea to have deterministic order
```

Comment:
Why do we need to have deterministic order? If we do, can't it be just an array?